clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -58,7 +59,7 @@
 
 public class HeartbeatRequestManagerTest {
```

Comment:
High level comment, I do see this test covering the timing logic for sending, and the response handling on error, but nothing for the successful HB response handling (important to ensure that it is updating the target assignment so that it can be processed by other components). Also it would be helpful to have some tests around HB timeouts, mainly to validate the retry logic around that. 
(Just suggestions for better coverage of core actions, OK for me if we prefer to target that in a separate PR)

Reply:
I added a test for the heartbeat timeout. See `testBackoffOnHeartbeatTimeout`